### PR TITLE
add vnu-validator 23.4.11

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
  - > I was not able to either build this from source, or find a feasible way to unravel all of the dependencies, much less their LICENSE texts 
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
  - > the `npmjs.org` tarball was the simplest way to get just the `.jar`, and not a whole standalone app with bundled jdk
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by bollwyvl** [comment]
@conda-forge/staged-recipes @conda-forge/help-java ready for review

File: recipes/vnu-validator/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  requires:
+    - m2-grep  # [win]
+    - python
+    - pytest
+
+about:
+  home: https://validator.github.io/validator
+  license: MIT
+  # TODO: see if licenses are in jar (unlikely)
+  license_file: LICENSE
```
Comment (xhochy) [inline recipes/vnu-validator/meta.yaml:31]: At least the git repo has something for thirdparties. 
```suggestion
  license_file: 
    - LICENSE
    - CREDITS
```

**Comment by bollwyvl** [comment]
@xhochy of _course_ `CREDITS` is not in the npm package, grumble, grumble...

**Comment by bollwyvl** [comment]
@xhochy back to :green_heart:

**Comment by bollwyvl** [comment]
@conda-forge/staged-recipes @conda-forge/help-java still looking :heavy_check_mark: here :hugs:

File: recipes/vnu-validator/conda_build_config.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+noarch_platform:
+  - unix  # [osx]
+  - unix  # [linux]
+  - win  # [win]
```
Comment (xhochy) [inline recipes/vnu-validator/conda_build_config.yaml:4]:
```suggestion
  - linux_64
  - win_64
```

**Review by xhochy — CHANGES_REQUESTED** [review]
`noarch_platform` work differently. They specify the OS where the CI is run on.

File: recipes/vnu-validator/conda_build_config.yaml (hunk 2 above)
Comment (bollwyvl) [reply recipes/vnu-validator/conda_build_config.yaml:4]: thanks for the review!,

we've got that over in `conda-forge.yml`: here, it's being used as the name of `__{virtual platform}`, as we nly want to end up with 2 builds (`unix` and `win`).

This is adapted from `ipython-feedstock`:
- https://github.com/conda-forge/ipython-feedstock/blob/main/recipe/conda_build_config.yaml
- https://github.com/conda-forge/ipython-feedstock/blob/main/conda-forge.yml#L18

Comment (isuruf) [reply recipes/vnu-validator/conda_build_config.yaml:4]: That's overly complicated. Here's a simplification of the ipython-feedstock. https://github.com/conda-forge/ipython-feedstock/pull/205

Comment (bollwyvl) [reply recipes/vnu-validator/conda_build_config.yaml:4]: yeah, tried that locally, getting:

```
recipes/vnu-validator/ has some lint:
  `noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: generic`.
```

File: recipes/vnu-validator/conda-forge.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+noarch_platforms:
```
Comment (bollwyvl) [inline recipes/vnu-validator/conda-forge.yml:1]: didn't know you could do that on `staged-recipes` :star2:

**Comment by bollwyvl** [comment]
Hm, now it looks like it's pulling some python metadata:

```
{
  "vnu-validator-23.4.11-h4f82c71": {
    "recipe": {
      "__unix": "__unix",
      "channel_targets": "conda-forge main",
      "python": "3.11.* *_cpython"
    }
  }
}
```

i guess if that's _just_ the hash, that's fine, but hope my choice of scripting language didn't have a negative effect on the installability of this...

File: recipes/vnu-validator/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+requirements:
+  run:
+    - __unix  # [unix]
+    - __win   # [win]
+    - openjdk >=8
+
+test:
+  requires:
+    - m2-grep  # [win]
+    - python
```
Comment (isuruf) [inline recipes/vnu-validator/meta.yaml:29]:
```suggestion
    - python *
```

**Comment by bollwyvl** [comment]
> `"python": "3.11.* *_cpython"`

Do I need to add some `ignore_run_exports` or something?

**Comment by bollwyvl** [comment]
lovely:

```
{
  "vnu-validator-23.4.11-h707e725": {
    "recipe": {
      "__unix": "__unix",
      "channel_targets": "conda-forge main"
    }
  }
}
```

**Comment by bollwyvl** [comment]
Welp, I have no further plans to change anything! Thanks for all the help!

**Review by xhochy — APPROVED** [review]

**Comment by bollwyvl** [comment]
Thanks all!

**Pull request merged**